Aquarius/Src/Aquarius/Events/KeyboardEvent.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include "Event.h"
+
+#include <GLFW/glfw3.h>
+
+
+namespace Aquarius {
+
+    class KeyEvent : public Event
+    {
+    public:
+
+        int getCode()
+        {
+            return m_keyCode;
+        }
+
+    protected:
+        int m_keyCode;
+
+        KeyEvent(int keyCode) : m_keyCode(keyCode) {}
+    };
+
+    class KeyPressedEvent : public KeyEvent
+    {
+    public:
+        KeyPressedEvent(int keyCode) : KeyEvent(keyCode) {}
+
+        EVENT_TYPE(KeyPressedEvent)
+
+        void logEvent()
+        {
+            AQ_CORE_TRACE("KeyPressedEvent: %v",m_keyCode);
+        }
+    };
+
+    class KeyRepeatedEvent : public KeyEvent
+    {
+    public:
+        KeyRepeatedEvent(int keyCode, int repeatCount) : KeyEvent(keyCode), m_repeatCount(repeatCount) {}
+
+        EVENT_TYPE(KeyRepeatedEvent)
+
+        void logEvent()
+        {
+            AQ_CORE_TRACE("KeyRepeatedEvent: %v", m_keyCode);
+        }
+
+    protected:
+        int m_repeatCount;
+    };
```

Comment:
There is no way for an external caller to access this member because it is protected. I doubt this will be subclassed so repeat count could be private (with a getter to provide access).